src/crl.c

```diff
@@ -99,8 +99,25 @@ static int InitCRL_Entry(CRL_Entry* crle, DecodedCRL* dcrl, const byte* buff,
         return WOLFSSL_FAILURE;
     }
 #endif
-    crle->certs = dcrl->certs;   /* take ownsership */
+#ifdef CRL_USE_CERT_BUFFERS
+    int totalCerts = dcrl->totalCerts;
+    RevokedCert* rc = dcrl->certs;
+
+    if (totalCerts >= CRL_MAX_REVOKED_CERTS)
+        return WOLFSSL_FAILURE;
+
+    while (totalCerts > 0) {
+        /* revoked cert chains were placed in reverse order in GetRevoked()
+         * copy the first cert in chain into the last array position */
+        if (rc != NULL) {
+            XMEMCPY(&crle->certs[--totalCerts], rc, sizeof(RevokedCert));
```

Comment:
Consider moving the `--totalCerts` out of XMEMCPY call for code clarity.

Reply:
@dgarske, Thanks for the review!  Implemented some of the suggested changes. 
Adding additional  `CRL_MAX_REVOKED_CERTS` check in the places you mentioned would be redundant. 
` if (dcrl->totalCerts >= CRL_MAX_REVOKED_CERTS)` is already checked in `GetRevoked()` when parsing a CRL.
I cannot see how `totalCerts` would ever be `> CRL_MAX_REVOKED_CERTS` in `CRL_Entry`. Let me know if I missed something.

Reply:
I’m thinking of a case where the buffer might not be initialized or was corrupted. The check you are referring to is a different function call. I think having the duplicate check makes safer code

Reply:
@dgarske, ok. got it.